### PR TITLE
Parental Leave Application: Bring in previous calculation slider for the durations screen

### PR DESCRIPTION
**Pull request by madebynoam**
## What

Check in the slider for the Parental Leave duration screen.
Refactor the Usage and Duration sliders into one component.

## Why

To continue building out the UX flow for parental leave.

## Screenshots / Gifs

image

## Known issues / Todo:
There is a small bug where half months can render an extra cell on the bar, this will be fixed in the next PR.
We will need to use localised strings in the Usage and Duration screens.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against master before asking for a review

### CODE REVIEW

**Review by arnifreyrs — COMMENTED** [review]
just one comment really

File: libs/application/templates/parental-leave/src/fields/Duration/index.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import { theme } from '@island.is/island-ui/theme'
+
+const ParentalLeaveUsage: FC<FieldBaseProps> = ({ field, formValue }) => {
+  const { id } = field
+  const { clearErrors } = useFormContext()
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+  // @ts-ignore
+  const monthsToUse = (formValue.usage as number) || 1
```
[the remark this answers is not in the document]
Comment (madebynoam) [reply libs/application/templates/parental-leave/src/fields/Duration/index.tsx:16]: Agree, I'll sync up with you on fixing this in a future PR

File: libs/application/templates/parental-leave/src/fields/Duration/index.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                index < monthsToUse
+              return {
+                background: isActive
+                  ? theme.color.mint400
+                  : theme.color.dark200,
+              }
+            }}
+            showMinMaxLabels
+            showToolTip
+            label={{ singular: 'mánuður', plural: 'mánuðir' }}
```
[the remark this answers is not in the document]
Comment (madebynoam) [reply libs/application/templates/parental-leave/src/fields/Duration/index.tsx:49]: Yeah I'll do a localisation pass after this PR.

File: libs/application/templates/parental-leave/src/fields/Duration/index.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+const ParentalLeaveUsage: FC<FieldBaseProps> = ({ field, formValue }) => {
+  const { id } = field
+  const { clearErrors } = useFormContext()
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+  // @ts-ignore
+  const monthsToUse = (formValue.usage as number) || 1
+  const [monthsToSpread, setMonthsToSpread] = useState<number | undefined>(
+    undefined,
```
[the remark this answers is not in the document]
Comment (madebynoam) [reply libs/application/templates/parental-leave/src/fields/Duration/index.tsx]: Thanks for the note, I've simplified this and fixed a bug where slider cells were showing as active if you moved the slider to less than the chosen usage. 

I still need to fix the half cell issue (where if it's a half month value it draws the whole cell as active), will be fixed in the next PR.

**Review by arnifreyrs — APPROVED** [review]
awesome!

**Pull request merged**